mission_control/navigator_missions/nav_missions/pinger.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+from __future__ import division
+import txros
+import std_srvs.srv
+import numpy as np
+import tf
+import tf.transformations as trns
+from navigator_msgs.msg import ShooterDoAction, ShooterDoActionGoal
+from navigator_msgs.srv import CameraToLidarTransform, CameraToLidarTransformRequest, FindPinger, FindPingerRequest, SetFrequency, SetFrequencyRequest
+from geometry_msgs.msg import Point
+from std_srvs.srv import SetBool, SetBoolRequest
+from twisted.internet import defer
+from image_geometry import PinholeCameraModel
+from visualization_msgs.msg import Marker,MarkerArray
+import navigator_tools
+
+class PingerMission:
+    OBSERVE_DISTANCE_METERS = 12
+    GATE_CROSS_METERS = 10
+    FREQ = 35000
+    LISTEN_TIME = 15
+
+    def __init__(self, navigator):
+        self.navigator = navigator
+        self.listen_client = self.navigator.nh.get_service_client('/hydrophones/set_listen', SetBool)
+        self.pinger_client = self.navigator.nh.get_service_client('/hydrophones/find_pinger', FindPinger)
+        self.reset_client = self.navigator.nh.get_service_client('/hydrophones/set_freq', SetFrequency)
+        self.markers_pub = self.navigator.nh.advertise("/pinger/debug_marker",MarkerArray)
+        self.negate = False
+
+    @txros.util.cancellableInlineCallbacks
```

Comment:
Like you did for the other ones, you should just return the defer instead of yielding on it inside of this function